### PR TITLE
Add argparse to FoV application

### PR DESCRIPTION
**Pull request by orelgueta**
Simple one for now

### CODE REVIEW

File: applications/validate_camera_fov.py [hunk 1]
```diff
@@ -12,25 +13,45 @@
...
+        default='south'
+    )
+
+    args = parser.parse_args()
 
-    site = 'south'
-    telescope = 'lst'
-    version = 'prod4'
-    label = 'lst-test'
+    label = '{}-test'.format(args.tel_type.lower())
```
Comment (RaulRPrado) [inline applications/validate_camera_fov.py:44]: All the output files (e.g. from the telescope_model) already contain the telescope type on the name, so it's redundant to add it to the label. I would use a label that only refers to the application itself. For the derive_mirror_rnda, I used derive_rnda.

File: applications/validate_camera_fov.py [hunk 2]
```diff
@@ -47,5 +68,7 @@
 
     # Now plot the camera as well
     plt = camera.plotPixelLayout()
-    plt.savefig('pixelLayout-LST.pdf', format='pdf', bbox_inches='tight')
+    cameraPlotFile = 'pixelLayout-{}.pdf'.format(telModel.telescopeType)
+    plt.savefig(cameraPlotFile, bbox_inches='tight')
```
Comment (RaulRPrado) [inline applications/validate_camera_fov.py:72]: Note for the future: We need to figure out a system to handle the application plots.

File: simtools/model/camera.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import matplotlib.pyplot as plt
 import matplotlib.patches as mpatches
 import matplotlib.colors as mcolors
-import scipy.spatial.cKDTree as KDTree
+from scipy.spatial import cKDTree as KDTree
```
Comment (RaulRPrado) [inline simtools/model/camera.py:7]: That was the bug that I added and fixed later. It was fixed in the master, though.

File: applications/validate_camera_fov.py (hunk 1 above)
Comment (orelgueta) [reply applications/validate_camera_fov.py:44]: Oh, OK, changed it now to `label = 'validate-FoV'`.

File: simtools/model/camera.py (hunk 3 above)
Comment (orelgueta) [reply simtools/model/camera.py:7]: Yes, but when I fixed it in my branch and then asked to pull it wanted me to commit first etc. and I was lazy, so I figure the merge would take care of it instead.

**Comment by orelgueta** [comment]
I guess now we can merge?

File: applications/validate_camera_fov.py (hunk 2 above)
Comment (orelgueta) [reply applications/validate_camera_fov.py:72]: Yes, we should find a solution. I don't know if the DB will provide a solution to this, it probably needs to be produced locally and optionally be pushed to the DB. Requires a bit more thinking I guess...

**Comment by RaulRPrado** [comment]
Yep

File: applications/validate_camera_fov.py (hunk 1 above)
Comment (GernotMaier) [reply applications/validate_camera_fov.py]: Can you explain to me what 'label' is doing / how it is used?

Comment (orelgueta) [reply applications/validate_camera_fov.py]: It is not relevant for this application. It is used when you compare one model to another. Most common use case is when you want to compare a modified model (lets say with modified random reflection angle) to one in the database (e.g., Prod4). In that case the label will be used in the plot comparing between the two (and in other places I guess, @RaulRPrado can expand I am sure).

Comment (RaulRPrado) [reply applications/validate_camera_fov.py]: Label is actually always relevant.
It's used to label all the output files, and the output directory structure. It's suppose to be used by all the modules that produce any kind of output.
There is the option of not giving any label (the date will be used for the output directory name) but it's not a good practice.

**Comment by GernotMaier** [comment]
When you say output directory, you mean for what?

The FOV produced a pdf in the working directory.

> On 25. May 2020, at 13:48, RaulRPrado <notifications@github.com> wrote:
> 
> 
> @RaulRPrado commented on this pull request.
> 
> In applications/validate_camera_fov.py:
> 
> >  
> -    site = 'south'
> -    telescope = 'lst'
> -    version = 'prod4'
> -    label = 'lst-test'
> +    label = '{}-test'.format(args.tel_type.lower())
> 
> Label is actually always relevant.
> It's used to label all the output files, and the output directory structure. It's suppose to be used by all the modules that produce any kind of output.
> There is the option of not giving any label (the date will be used for the output directory name) but it's not a good practice.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 



---
Dr Gernot Maier
CTA
Deutsches Elektronen-Synchrotron DESY 
Ein Forschungszentrum der Helmholtz-Gemeinschaft
Platanenallee 6, D-15738 Zeuthen, Germany

Tel.: +49  33 7627 7598
Internet: www.desy.de/cta
Besucheradresse: 1X07

**Comment by RaulRPrado** [comment]
> When you say output directory, you mean for what? The FOV produced a pdf in the working directory.

You can ask the TelescopeModel to exportConfigFile, for example. Or you can ask the Camera to export a new camera config file with a certain change (not implemented, just an example).
This is a peculiar application that does not produce any output, but basically all the other ones do.

The output directory is split. If you run any ray tracing application, there will be model related outputs and ray tracing related outputs, for example.

**Comment by orelgueta** [comment]
And in particular we haven't figured out yet what to do with output plots (as we mentioned at some point today in one of these many issues). We think they should be produced locally and then optionally be pushed into the database.

**Comment by GernotMaier** [comment]
ok!

> On 25. May 2020, at 14:01, RaulRPrado <notifications@github.com> wrote:
> 
> 
> When you say output directory, you mean for what? The FOV produced a pdf in the working directory.
> 
> You can ask the TelescopeModel to exportConfigFile, for example. Or you can ask the Camera to export a new camera config file with a certain change (not implemented, just an example).
> This is a peculiar application that does not produce any output, but basically all the other ones do.
> 
> The output directory is split. If you run any ray tracing application, there will be model related outputs and ray tracing related outputs, for example.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 



---
Dr Gernot Maier
CTA
Deutsches Elektronen-Synchrotron DESY 
Ein Forschungszentrum der Helmholtz-Gemeinschaft
Platanenallee 6, D-15738 Zeuthen, Germany

Tel.: +49  33 7627 7598
Internet: www.desy.de/cta
Besucheradresse: 1X07

**Pull request merged**